### PR TITLE
Update README for 0.54

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -79,10 +79,14 @@ https://numba.readthedocs.io/en/stable/index.html
 Mailing Lists
 =============
 
-Join the Numba mailing list numba-users@continuum.io:
-https://groups.google.com/a/continuum.io/d/forum/numba-users
```
Comment (sklam) [inline README.rst:83]: oh... didn't kn we still have that link here

**Review by sklam — APPROVED** [review]
LGTM

**Comment by esc** [comment]
Thanks for the patch! Looks great!

File: README.rst (hunk 1 above)
Comment (stuartarchibald) [reply README.rst:83]: Only just realised today!

**Pull request merged**